AntennaTracker/system.cpp

```diff
@@ -231,10 +231,7 @@ bool Tracker::set_mode(const uint8_t new_mode, const ModeReason reason)
  */
 bool Tracker::should_log(uint32_t mask)
 {
-    if (!logger.should_log(mask)) {
-        return false;
-    }
-    return true;
+    return logger.should_log(mask);
```

Comment:
This is moving us away from a very common pattern in ArduPilot; a lot of our large boolean functions have this pattern of if-clause-returns.  Check AP_Arming.cpp in any vehicle, for example.

I'm not keen on making the last clause special like you are in this patch. It means if someone wants to add a clause they may have to restructure it back the way it was.  And then someone else wants to change it back to this if that clause moves up into common....

It also removes the possibility to comment separately on the two cases being taken.  Something missing from a lot of these blocks...

It's not a simplification.  It breaks the pattern in the rest of the function (in larger cases).  I'd prefer not to have this particular sort of change.  I haven't commented the others in this PR, but they do stand out.

Reply:
On this one I'd disagree, I think the proposed version is actually simpler and more straightforward. It's starting to be a style thing though, where changing at this point seems more like noise then worth time.